client/src/pages/LogIn_SignUp.js

```diff
@@ -0,0 +1,208 @@
+  Typography,
+  Link,
+  Paper,
+  Grid,
+  Box,
+  makeStyles,
+} from '@material-ui/core';
+import React, { useEffect, useState, useCallback } from 'react';
+import { useLocation } from 'react-router-dom';
+
+const useStyles = makeStyles({
+  textField: {
+    width: '50%',
+  },
+  bold: {
+    color: 'black',
+  },
+  seperator: {
+    width: '10%',
+    border: '1px solid #00e676',
+  },
+  formPadding: {
+    padding: '20px',
+  },
+});
+
+const LogIn_SignUp = () => {
+  //state declaration
+  const [signInEmail, setEmail] = useState('');
+  const [signInPassword, setPassword] = useState('');
+  const [registerName, setName] = useState('');
+  const [confirmPassword, setConfirmPassword] = useState('');
+  const [lengthValidation, setLengthValidation] = useState(true);
+  const [confirmValidation, setConfirmValidation] = useState(true);
+
+  const classes = useStyles();
+
+  //Determines which form elements to render based on the path
+  const signInForm = useLocation().pathname === '/login' ? true : false;
+
+  //change this to change the min length requirement
+  const minLength = 6;
+
+  //front-end validation for registering users
+  const handleValidation = useCallback(() => {
+    if (signInPassword !== '' && signInPassword.length < minLength) {
+      setLengthValidation(false);
+    } else {
+      setLengthValidation(true);
+    }
+    if (confirmPassword !== '' && signInPassword !== confirmPassword) {
+      setConfirmValidation(false);
+    } else {
+      setConfirmValidation(true);
+    }
+  }, [signInPassword, confirmPassword]);
+
+  useEffect(() => {
+    handleValidation();
+  }, [signInPassword, confirmPassword, handleValidation]);
+
+  const handleChange = (event) => {
+    const { value, name } = event.target;
+    if (name === 'signInEmail') {
+      setEmail(value);
+      console.log(signInEmail);
+    } else if (name === 'signInPassword') {
+      setPassword(value);
+    } else if (name === 'registerName') {
+      setName(value);
+    } else if (name === 'confirmPassword') {
+      setConfirmPassword(value);
+    }
+  };
+
+  const handleSubmit = (event) => {
+    event.preventDefault();
+    setEmail('');
+    setPassword('');
+    setName('');
+    setConfirmPassword('');
+    setLengthValidation(true);
+    setConfirmValidation(true);
+  };
+
+  return (
+    <Grid container>
+      <Grid item xs></Grid>
+      <Grid item xs={8} sm={6} md={4} p={2}>
+        <Box pt={4} textAlign="center">
+          <Paper elevation={3}>
+            <form onSubmit={handleSubmit} className={classes.formPadding}>
+              <Box>
+                <Typography variant="h4">
+                  {signInForm ? (
+                    <legend>Sign In</legend>
+                  ) : (
+                    <legend>Sign Up</legend>
+                  )}
+                  <hr className={classes.seperator} />
+                </Typography>
+              </Box>
+              <Box mt={4}>
+                {signInForm ? null : (
```

Comment:
In cases like this you can use the `&&` operator to avoid having to return null

Reply:
Removed the check all together since I have separated the forms into their own pages